src/utils/transform-filters.ts

```diff
@@ -0,0 +1,74 @@
+import { CombinedFilter, Filter, FilterCombinator } from '@yext/answers-core';
+import { SelectableFilter } from '../models/utils/selectablefilter';
+
+/**
+ * Combine a given list of Filter using OR condition into a single CombinedFilter object
+ */
+function combineFiltersWithOR(filters: Filter[]): Filter | CombinedFilter {
+  if (filters.length === 1) {
+    return filters[0];
+  }
+  return {
+    combinator: FilterCombinator.OR,
+    filters: filters
+  };
+}
+
+/**
+ * Convert a list of SelectableFilter to nested filter structure use in core
+ */
+function transformAFilterSetToCoreFormat(
+  filters: SelectableFilter[]
+): Filter | CombinedFilter | null {
+  if (filters.length === 0) {
+    return null;
+  }
+  if (filters.length === 1) {
+    return { ...filters[0].filter };
+  }
+  const groupedFilters: Record<string, Filter[]> = filters.reduce((groups, element) => {
+    groups[element.filter.fieldId]
+      ? groups[element.filter.fieldId].push({ ...element.filter })
+      : groups[element.filter.fieldId] = [{ ...element.filter }];
+    return groups;
+  }, {});
+
+  const groupedFilterLabels = Object.keys(groupedFilters);
+  if (groupedFilterLabels.length === 1) {
+    return combineFiltersWithOR(groupedFilters[groupedFilterLabels[0]]);
+  }
+  return {
```

Comment:
Assuming this behavior of ORing together filters with the same fileldId and ANDing together the filters grouped by filterId is the same as the SDK, this logic LGTM

Reply:
I followed the logic we had for previous StaticFilters in React. But I just double check the SDK stuff (https://github.com/yext/answers-search-ui/blob/master/src/ui/components/filters/filteroptionscomponent.js#L719-L726) seems like the logic is the same so we should be good